dask_expr/_expr.py

```diff
@@ -1945,11 +1978,21 @@ class BlockwiseHead(Head, Blockwise):
     the first `n` rows of an entire collection.
     """
 
+    _parameters = ["frame", "n", "npartitions", "safe"]
+
+    @functools.cached_property
+    def npartitions(self):
+        return len(self._divisions()) - 1
+
     def _divisions(self):
-        return self.frame.divisions
+        return self.frame.divisions[: len(self._partitions) + 1]
 
     def _task(self, index: int):
-        return (M.head, (self.frame._name, index), self.n)
+        if self.safe:
+            op = safe_head
+        else:
+            op = M.head
```

Comment:
nit, compacts down slightly nicer.
```suggestion
        op = safe_head if self.safe else M.head
```

Reply:
I like spreading them out better here